### PR TITLE
add_alerting_system

### PR DESCRIPTION
**Pull request by GreatGaryCooper**
## What does this MR do?
We added an alerting agent, that is connected to a telegram bot right now.
We've added a dependency injection container library called "[inversify](https://github.com/inversify/InversifyJS)" to manage the alerting agent instance.

## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [x] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

**Comment by hypersolo** [comment]
noiceeee :fire: 
just a few things:
* add description to env.example for new env.variable
* update readme
* move alerting from data layer to utils
:love_letter:

File: src/utils/config/env-validation.ts [hunk 1]
```diff
@@ -306,4 +306,22 @@ export default {
     env: "SIGN_IN_MESSAGE",
     default: null,
   },
+  TELEGRAM_BOT_TOKEN: {
```
Comment (hypersolo) [inline src/utils/config/env-validation.ts:309]: alerter: {bot: string, channel: string, active: bool} might be a better interface.
mentioning telegram directly is not a good idea :love_letter:

File: .env.example [hunk 2]
```diff
@@ -132,5 +132,6 @@ ARBITRUM_PROVIDERS_URLS=https://arb-mainnet.g.alchemy.com/v2/YOUR_ALCHEMY_PROJEC
 # ---------------------------------------------------------------------
 
 
-
-
+ALERT_BOT_ACTIVE=false
```
Comment (hypersolo) [inline .env.example:135]: add a small description bro :beers:
and also update readme

File: src/utils/config/env-validation.ts (hunk 1 above)
Comment (GreatGaryCooper) [reply src/utils/config/env-validation.ts:309]: Alerter is going to have many interfaces for different messaging platforms, like telegram and discord.
For example discord probably is going to have an additional parameter called `server_id` alongside the `channel_id`.
So environment variables vary based on the platform used for alerter.
Is it clear? 🤔

**Review by hypersolo — APPROVED** [review]

**Comment by WilliamHawkingStephen** [comment]
- nice one broski :heart:

**Review by WilliamHawkingStephen — APPROVED** [review]

**Pull request merged**